ITensorGaussianMPS/src/linalg.jl

```diff
@@ -0,0 +1,201 @@
+"""
+Some of the functionality in this script is closely related to routines in the following package
+https://github.com/Jacupo/F_utilities
+and the associated publication
+10.21468/SciPostPhysLectNotes.54
+"""
+
+"""Takes a single-particle Hamiltonian in blocked Dirac format and finds the fermionic transformation U that diagonalizes it"""
+function _gaussian_eigen_blocked(H; pert=nothing)
+  #make sure H is Hermitian
+  @assert ishermitian(H)
+  H = Hermitian(H)
+  ElT = eltype(H)
+  #convert from Dirac to Majorana picture
+  N = size(H, 1)
+  Ω = build_Ω(ElT, N)
+  h = real(-im .* (Ω * H * Ω'))
+  h = (h - h') ./ 2
+  #@show size(h)
+  if !isnothing(pert)
+    noise = rand(size(h)...) * pert
+    noise = (noise - noise') ./ 2
+    h = h + noise
+  end
+  # Schur diagonalize including reordering
+  _, O, vals = order_schur(schur(h))
+  # convert back to Dirac Frame
+  Fxpxx = build_Fxpxx(N)
+  U = Ω' * O * (Fxpxx') * Ω
+  d = vcat(-vals, vals)
+  if ElT <: Real
+    U = make_real_if_possible(U, d)
+    # make another pass with rotation in the complex plane per eigenvector
+    U .*= exp.(-im * angle.(U[1:1, :]))
+    @assert norm(imag.(U)) < sqrt(eps(real(ElT)))
+    U = real(real.(U))
+  end
+  return d, U
+end
+
+"""Takes a single-particle Hamiltonian in interlaced Dirac format and finds the complex fermionic transformation U that diagonalizes it."""
+function gaussian_eigen(H; pert=nothing)
```

Comment:
I would vote for naming this `eigen_gaussian` instead.

Reply:
Also to make the keyword argument more descriptive, I would call it something like `noise_scale` instead of `pert`.